File: 第十次/10-2/Rectangle.cpp
#include"Rectangle.h"

//建構子初值設定
Rectangle::Rectangle() {
	width = 1;
	height = 1;
}
//引入(double,double) 的建構子
Rectangle::Rectangle(double newWidth, double newHeight) {
	setWidth(newWidth);
	setHeight(newHeight);
}

double Rectangle::getWidth() {
	return width;
}
void Rectangle::setWidth(double newWidth) {
	width = newWidth;
}

double Rectangle::getHeight() {
	return height;
}
void Rectangle::setHeight(double newHeight) {
	height = newHeight;
}

double Rectangle::getArea() {
	return height * width;
}

File: Python/第10章/10-3/Rectangle.py
class Rectangle:
    
    numOfRect = 0

    # 未完成 static 
    def __init__(self, width, height):
        self.__width = width
        self.__height = 

    '''
    def __new__(cls):
        cls.numOfRect+=1
    '''
        
    def setWidth(self, width):
        self.__width = width
    def getWidth(self):
        return self.__width
    def setHeight(self, height):
        self.__height = height
    def getHeight(self):
        return self.__height
    def getArea(self):
        return self.__height * self.__width

    @classmethod
    def getNumOfRect(cls):
        return cls.numOfRect
